### PR TITLE
error for hybrid continuous/discrete systems

### PR DESCRIPTION
**Pull request by baggepinnen**
This PR detects hybrid continuous/discrete systems and throws an error indicating that they are not yet supported by the default MTK compiler, indicating that JuliaSimCompiler is currently required for correctness.
The tests are also deactivated.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -26,7 +26,7 @@ end
             @safetestset "Direct Usage Test" include("direct.jl")
             @safetestset "System Linearity Test" include("linearity.jl")
             @safetestset "Input Output Test" include("input_output_handling.jl")
-            @safetestset "Clock Test" include("clock.jl")
+            # @safetestset "Clock Test" include("clock.jl")
```
Comment (ChrisRackauckas) [inline test/runtests.jl:29]: we don't deactivate tests like this. Mark them as throwing the right error.

File: src/systems/systemstructure.jl [hunk 2]
```diff
@@ -636,6 +636,9 @@ function structural_simplify!(state::TearingState, io = nothing; simplify = fals
             check_consistency, fully_determined,
             kwargs...)
         if length(tss) > 1
+            if continuous_id > 0
+                error("Hybrid continuous-discrete systems are currently not supported with the standard MTK compiler. This system requires JuliaSimCompiler.jl, see https://help.juliahub.com/juliasimcompiler/stable/")
```
Comment (ChrisRackauckas) [inline src/systems/systemstructure.jl:640]: can we make this be a named error exception?

Comment (baggepinnen) [reply src/systems/systemstructure.jl:640]: what is a named error exception?

**Comment by ChrisRackauckas** [comment]
Lots of typos found: https://github.com/SciML/ModelingToolkit.jl/actions/runs/9384644106/job/25840889262?pr=2765

File: test/parameter_dependencies.jl [hunk 3]
```diff
@@ -81,28 +81,34 @@ end
...
+
+    @test_skip begin
+        Tf = 1.0
+        prob = ODEProblem(sys, [x => 0.0, y => 0.0], (0.0, Tf),
+            [kp => 1.0; z(k - 1) => 3.0; yd(k - 1) => 0.0; z(k - 2) => 4.0;
+             yd(k - 2) => 2.0])
+        @test_nowarn solve(prob, Tsit5(); kwargshandle = KeywordArgSilent)
+
+        @mtkbuild sys = ODESystem(eqs, t; parameter_dependencies = [kq => 2kp],
+            discrete_events = [[0.5] => [kp ~ 2.0]])
```
Comment (ChrisRackauckas) [inline test/parameter_dependencies.jl:95]: I thought directly defined events were fine? What's the issue here?

Comment (baggepinnen) [reply test/parameter_dependencies.jl:95]: The equations used here are still clocked, this test adds a manually defined discrete event _in addition_ to the clocked system

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Handle the typos first.

**Review by YingboMa — APPROVED** [review]

**Pull request merged**